src/utilities/createRouterReject.ts

```diff
@@ -0,0 +1,91 @@
+import { Ref, ref } from 'vue'
+import { NotFound } from '@/components'
+import { RegisteredRejection, Resolved, Route, RouteComponent } from '@/types'
+
+export const builtInRejections = ['NotFound'] as const
+export type BuiltInRejection = typeof builtInRejections[number]
+
+export const builtInRejectionComponents: Record<BuiltInRejection, RouteComponent> = {
+  NotFound,
+}
+
+export type RouterRejection = BuiltInRejection | RegisteredRejection
+
+type BuiltInRejectionComponents = Partial<Record<BuiltInRejection, RouteComponent>>
+
+export type RouterRejectionComponents = RegisteredRejection extends never
+  ? { rejections?: BuiltInRejectionComponents }
```

Comment:
just to clarify here, if the developer doesn't have any additional RegisteredRejections, the rejections property itself is optional. Presumably because the only rejection (NotFound) is handled but could be overridden. If the user does provide their own rejections, the rejections property is required. Any user defined rejections are required, the NotFound is still optional. 

this is beautiful

Reply:
Yep exactly. ❤️